src/main/java/fr/zcraft/zlib/components/gui/ArrayPromptGui.java

```diff
@@ -0,0 +1,93 @@
+package fr.zcraft.zlib.components.gui;
+
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import fr.zcraft.zlib.components.gui.ExplorerGui;
+import fr.zcraft.zlib.components.gui.Gui;
+import fr.zcraft.zlib.tools.Callback;
+
+/* Built from recommendations in PR https://github.com/zDevelopers/zLib/pull/14 */
+public class ArrayPromptGui<T> extends ExplorerGui<T> {
+
+	private Callback<T> cb;
+	private String purpose;
```

Comment:
This should be called `title` (clearer), I think. And all parameters and variables named the same way.